File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py

import sys

input = sys.stdin.readline

N, M = list(map(int, input().split()))

result = []

def backtrack(start, curr):
    if len(curr) == M:
        result.append(curr[:])
        return
    
    for i in range(start, N+1):
        curr.append(i)
        backtrack(i, curr)
        curr.pop()

backtrack(1, [])

for r in result:
    for c in r:
        print(c, end=" ")
    print()

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int(String($0))! }
let n = input[0]
let m = input[1]
var nums: [Int] = []

func recursive(cur: Int, start: Int) {
  if cur == m {
    for num in nums {
      print(num, terminator: " ")
    }
    print()
    return
  }

  for i in start..<n {
    nums.append(i+1)
    recursive(cur: cur + 1, start: i)
    nums.removeLast()
  }
}

recursive(cur: 0, start: 0)